### PR TITLE
Outreachy - added initial ideas

### PR DESCRIPTION
**Pull request by jpkrohling**
Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

### CODE REVIEW

**Comment by pritianka** [comment]
All looks good! I haven't had the time to grammar check yet but if you've done a spell-check etc, can merge since I know this is urgent.

**Comment by jpkrohling** [comment]
@objectiser , could you proof-read this one?

**Comment by reimertz** [comment]
@jpkrohling LGTM and great work! Just give me an 👍  and I'll merge it.

File: pages/outreachy.md [hunk 1]
```diff
@@ -54,8 +57,56 @@ Interested in becoming a mentor? Contact one of the coordinators!
...
+
+* Collect ideas on what's possible and what's desirable for mobile applications
+* Decide and experiment with different options for an optimized mobile experience, like, which sampling strategies
+make more sense
+* Create the missing pieces, like OpenTracing framework integration for some Android component, or a specific Jaeger
+tracer for mobile usage
+* Documentation + examples
+* The end result would be: a trace starts when a user opens the app on the phone and backend microservices are called
+
+### Split the JavaScript OT library: NodeJS and Browser
```
Comment (objectiser) [inline pages/outreachy.md]: This is a jaeger specific task I believe @pavolloffay ?

File: pages/outreachy.md [hunk 2]
```diff
@@ -54,8 +57,56 @@ Interested in becoming a mentor? Contact one of the coordinators!
...
+
+Mentor: Pavol Loffay
+
+* HttpSender which works in a web browser
+* Intercept web browser events (maybe can be done as OpenTracing instrumentation)
+
+### Drive consensus on a common format for reporting trace/span data
+
+Mentor: Gary Brown
+
```
Comment (objectiser) [inline pages/outreachy.md]: Would be good to reference this https://github.com/opentracing/specification/issues/86 - although it discusses both in-band and out-of-band, but it sets the context aswell.

File: pages/outreachy.md [hunk 3]
```diff
@@ -54,8 +57,56 @@ Interested in becoming a mentor? Contact one of the coordinators!
...
+* The end result would be: a trace starts when a user opens the app on the phone and backend microservices are called
+
+### Split the JavaScript OT library: NodeJS and Browser
+
+Mentor: Pavol Loffay
+
+* HttpSender which works in a web browser
+* Intercept web browser events (maybe can be done as OpenTracing instrumentation)
+
+### Drive consensus on a common format for reporting trace/span data
```
Comment (yurishkuro) [inline pages/outreachy.md]: I am not sure this is a good intern project, it requires a lot more experience with the tracing space

File: pages/outreachy.md [hunk 4]
```diff
@@ -54,8 +57,56 @@ Interested in becoming a mentor? Contact one of the coordinators!
 
 ## Available tasks
 
-Coming soon. At this moment, we recommend looking at the following issue trackers for smaller tasks that might
-give a sense of what's coming:
+### Instrumentation for mobile applications
```
Comment (yurishkuro) [inline pages/outreachy.md]: Doing mobile instrumentation efficiently is actually a non-trivial engineering project. For example, our mobile team has a whole _platform_ sub-team that built a _performance SDK_ that does clever things like gauging the available network bandwidth from the phone and sending monitoring data via unused bandwidth only.

Comment (objectiser) [reply pages/outreachy.md]: Agree that to provide a production quality solution will be non-trivial, but this type of project has been popular with GSoC students on Hawkular, and the expectation would only be to provide an initial proof of concept with demonstrator. If it then looks promising, further performance work could be done by more experienced contributors?

File: pages/outreachy.md (hunk 3 above)
Comment (objectiser) [reply pages/outreachy.md]: Reason I thought this might be good is that the intern project is just to document and implement prototype (one or more language tracers/jaeger backend adapter) - guidance on what should be in the standard format should come from the community, so their role would be to request input and feed that into their prototype. The initial starting point would be that it must capture the information provided via the OpenTracing API.
@bhs Thoughts?

File: pages/outreachy.md [hunk 5]
```diff
@@ -54,8 +57,56 @@ Interested in becoming a mentor? Contact one of the coordinators!
...
+Mentor: Pavol Loffay
+
+* For instance, for missing Android parts.
+
+### Write Jaeger tracer implementation for any OpenTracing API.
+
+Mentor: Gary Brown
+
+* C#
+* Kotlin for Android
```
Comment (pavolloffay) [inline pages/outreachy.md]: This is funny there is no OpenTracing Kotlin API

File: pages/outreachy.md (hunk 1 above)
Comment (pavolloffay) [reply pages/outreachy.md]: Indeed it is!

File: pages/outreachy.md (hunk 5 above)
Comment (jpkrohling) [reply pages/outreachy.md]: I believe that Kotlin is a JVM-based language, so, the Java one would work. I'm not sure Kotlin would make sense here, but it's part of the task to figure this out.

File: pages/outreachy.md (hunk 3 above)
Comment (jpkrohling) [reply pages/outreachy.md]: If we keep this, could you send a PR to change this page with the outcome of the discussion?

Comment (objectiser) [reply pages/outreachy.md]: @yurishkuro Would rewording as @bhs suggests be ok, or would you still prefer it is removed?

**Comment by pavolloffay** [comment]
A good task might be to write a test suite that validates tracer implementation.

File: pages/outreachy.md (hunk 1 above)
Comment (yurishkuro) [reply pages/outreachy.md]: Maybe add Jaeger in the title

File: pages/outreachy.md (hunk 3 above)
Comment (yurishkuro) [reply pages/outreachy.md]: Yes. Can we add some description?

**Comment by jpkrohling** [comment]
> A good task might be to write a test suite that validates tracer implementation.

You mean, like a test harness for OpenTracing implementations?

**Comment by pavolloffay** [comment]
yes
